src/PlusDataCollection/Andor/vtkPlusAndorCamera.cxx

```diff
@@ -0,0 +1,956 @@
+  versionString << "Andor SDK version: "  << SDKVersion << std::ends;
+
+  return versionString.str();
+}
+
+
+//----------------------------------------------------------------------------
+vtkPlusAndorCamera::vtkPlusAndorCamera()
+{
+  this->RequirePortNameInDeviceSetConfiguration = true;
+
+  this->StartThreadForInternalUpdates = false; // frames should not be acquired automatically
+  this->AcquisitionRate = 1.0; // this controls the frequency
+}
+
+// ----------------------------------------------------------------------------
+vtkPlusAndorCamera::~vtkPlusAndorCamera()
+{
+  if(!this->Connected)
+  {
+    this->Disconnect();
+  }
+}
+
+// ----------------------------------------------------------------------------
+PlusStatus vtkPlusAndorCamera::InitializeAndorCamera()
+{
+  checkStatus(Initialize(""), "Initialize");
+
+  // Check the safe temperature, and the maximum allowable temperature on the camera.
+  // Use the min of the two as the safe temp.
+  int MinTemp, MaxTemp;
+  unsigned result = checkStatus(GetTemperatureRange(&MinTemp, &MaxTemp), "GetTemperatureRange");
+  if(result == DRV_SUCCESS)
+  {
+    LOG_INFO("The temperature range for the connected Andor Camera is: " << MinTemp << " and " << MaxTemp);
+  }
+
+  if(MaxTemp < this->SafeTemperature)
+  {
+    this->SafeTemperature = MaxTemp;
+  }
+  if(this->CoolTemperature < MinTemp || this->CoolTemperature > MaxTemp)
+  {
+    LOG_ERROR("Requested temperature for Andor camera is out of range");
+    return PLUS_FAIL;
+  }
+
+  int x, y;
+  checkStatus(GetDetector(&x, &y), "GetDetector");
+  frameSize[0] = static_cast<unsigned>(x);
+  frameSize[1] = static_cast<unsigned>(y);
+
+  // init to binning of 1 (meaning no binning), and full sensor size
+  checkStatus(SetImage(this->HorizontalBins, this->VerticalBins, 1, x, 1, y), "SetImage");
+
+  checkStatus(PrepareAcquisition(), "PrepareAcquisition");
+
+  return PLUS_SUCCESS;
+}
+
+// ----------------------------------------------------------------------------
+void vtkPlusAndorCamera::InitializePort(DataSourceArray& port)
+{
+  for(unsigned i = 0; i < port.size(); i++)
+  {
+    port[i]->SetPixelType(VTK_UNSIGNED_SHORT);
+    port[i]->SetImageType(US_IMG_BRIGHTNESS);
+    port[i]->SetOutputImageOrientation(US_IMG_ORIENT_MF);
+    port[i]->SetInputImageOrientation(US_IMG_ORIENT_MF);
+    port[i]->SetInputFrameSize(frameSize);
+
+    LOG_INFO("Andor source initialized. ID: " << port[i]->GetId());
+  }
+}
+
+// ----------------------------------------------------------------------------
+PlusStatus vtkPlusAndorCamera::InternalConnect()
+{
+  LOG_TRACE("vtkPlusAndorCamera::InternalConnect");
+  if(this->InitializeAndorCamera() != PLUS_SUCCESS)
+  {
+    LOG_ERROR("Andor camera failed to initialize.");
+    return PLUS_FAIL;
+  }
+
+  this->GetVideoSourcesByPortName("BLIraw", BLIraw);
+  this->GetVideoSourcesByPortName("BLIrectified", BLIrectified);

```

Comment:
Should the corrected ports be called BLICorrected and GrayCorrected or BLIProcessed and GrayProcessed? I think "rectified" implies an operation that is not happening.